aiida_mlip/data/model.py

```diff
@@ -0,0 +1,220 @@
+
+        Returns
+        -------
+        Path
+            The path of the model file of interest (same as input path if no duplicates were found).
+        """
+        file_hash = cls._calculate_hash(file)
+
+        def is_diff_file(curr_path: Path):
+            return curr_path.is_file() and not curr_path.samefile(file)
+
+        file_folder = Path(file).parent
+        for existing_file in filter(is_diff_file, file_folder.rglob("*")):
+            if cls._calculate_hash(existing_file) == file_hash:
+                file.unlink()
+                return Path(existing_file)
+        return Path(file)
+
+    def __init__(
+        self,
+        file: Union[str, Path],
+        architecture: str,
+        filename: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the ModelData object.
+
+        Parameters
+        ----------
+        file : Union[str, Path]
+            Absolute path to the file.
+        architecture : [str]
+            Architecture of the mlip model.
+        filename : Optional[str], optional
+            Name to be used for the file (defaults to the name of provided file).
+
+        Other Parameters
+        ----------------
+        kwargs : Any
+            Additional keyword arguments.
+        """
+        super().__init__(file, filename, **kwargs)
+        self.base.attributes.set("architecture", architecture)
+
+    def set_file(
+        self,
+        file: Union[str, Path],
+        filename: Optional[str] = None,
+        architecture: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Set the file for the node.
+
+        Parameters
+        ----------
+        file : Union[str, Path]
+            Absolute path to the file.
+        architecture : [str]
+            Architecture of the mlip model.
+        filename : Optional[str], optional
+            Name to be used for the file (defaults to the name of provided file).
+
+
+        Other Parameters
+        ----------------
+        kwargs : Any
+            Additional keyword arguments.
+        """
+        super().set_file(file, filename, **kwargs)
+        self.base.attributes.set("architecture", architecture)
+
+    @classmethod
+    def local_file(
+        cls,
+        file: Union[str, Path],
+        architecture: str,
+        filename: Optional[str] = None,
+    ):
+        """Create a ModelData instance from a local file.
+
+        Parameters
+        ----------
+        file : Union[str, Path]
+            Path to the file.
+        architecture : [str]
+            Architecture of the mlip model.
+        filename : Optional[str], optional
+            Name to be used for the file (defaults to the name of provided file).
+
+        Returns
+        -------
+        ModelData
+            A ModelData instance.
+        """
+        file_path = Path(file).resolve()
+        return cls(file=file_path, architecture=architecture, filename=filename)
+
+    @classmethod
```

Comment:
I'm curious why this and `check_existing_file` are `classmethod`s when they seem to depend only on aspects of self.